### ISSUE TITLE
PubSup close due to buffer size is not handled

### ISSUE THREAD

**Issue by SplotyCode** [redisson/redisson#3923]
**Expected behavior**
Redisson should notify about closed subscriptions
**Actual behavior**
The subscription is closed quitely mem dump of redisson shows that the pubsubs are even still in PublishSubscribeService maps.
**Steps to reproduce or test case**
```
public class Main {
    public static void main(String[] args) {
        RedissonClient client = Redisson.create();
        RTopic topic = client.getTopic("abc");
        topic.addListener(new StatusListener() {
            @Override
            public void onSubscribe(String channel) {
                System.out.println("subscribe");
            }

            @Override
            public void onUnsubscribe(String channel) {
                System.out.println("unsubscribe");
            }
        });
        topic.addListener(UUID.class, (channel, msg) -> {
            System.out.println("Got message " + channel + " " + Thread.currentThread().getName());
        });
        while (true) {
            client.getBucket("set-" + Math.random()).setAsync(UUID.randomUUID()).onComplete((unused, throwable) -> {
                System.out.println(Thread.currentThread().getName());
                try {
                    Thread.sleep(15_000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            });
            try {
                Thread.sleep(80);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}
```
```
public class Main2 {
    public static void main(String[] args) throws InterruptedException {
        for (int i = 0; i < 10; i++) {
            RedissonClient client = Redisson.create();
            RTopic topic = client.getTopic("abc");
            new Thread(() -> {
                while (true) {
                    topic.publishAsync(builder());
                    Uninterruptibles.sleepUninterruptibly(80, TimeUnit.MILLISECONDS);
                }
            }).start();

        }
    }

    private static String builder() {
        StringBuilder builder = new StringBuilder();
        for (int i = 0; i < 2_000; i++) {
            builder.append(UUID.randomUUID());
        }
        return builder.toString();
    }
}
```

**Redis version**
5.0.3
**Redisson version**
3.16.4
**Redisson configuration**
Single node with UseThreadClassLoader=false

**Comment by mrniko** [comment]
I run both classes simultaneously. Output of `redisson-netty-*` line is ongoing. Subscription works

**Comment by avcad** [comment]
I don't think the problem is specific to the buffers scenario.  Even running only Main(), the re-connect/re-subscribe takes  a while because he is using all the netty threads with the sleep.   Thread.sleep(15_000);
so I ran Main() only, resubscribe took a long time. then I changed the sleep to 100, and it happened right away. I think the issue is just that the netty pool is full, and this should affect all operations. I think it's a better practice to do the slow work somewhere else
